### PR TITLE
Streamline dev dependency management

### PR DESCRIPTION
**Pull request by jakob-keller**
### Description of Change
Prepares CI/CD workflow for testing of distribution packages (#1169)

### Assumptions
* `aiohttp` version constraint in CI/CD workflow is no longer required
* `time` statements in CI/CD workflow are no longer required

### Checklist for All Submissions
* [ ] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* [ ] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details):
  * [ ] Detailed description of issue
  * [ ] Alternative methods considered (if any)
  * [ ] How issue is being resolved
  * [ ] How issue can be reproduced
* [ ] If this is providing a new feature  (can be provided via link to issue with these details):
  * [ ] Detailed description of new feature
  * [ ] Why needed
  * [ ] Alternatives methods considered (if any)

### Checklist when updating botocore and/or aiohttp versions

* [ ] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [ ] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [ ] I have ensured that the awscli/boto3 versions match the updated botocore version

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # generated by users and intentionally not pushed to the repo
```
Comment (webknjaz) [inline .gitignore:1]: This is weird. pip-tools based workflows are specifically built around having "lock files" checked in for reproducibility..

**Comment by webknjaz** [comment]
What's the game plan here? This is definitely different from anything I've ever seen in the wild…

File: .gitignore (hunk 1 above)
Comment (thehesiod) [reply .gitignore:1]: to make this legit @jakob-keller we should have one lockfile per python version and delete this cmt and txt

**Comment by thehesiod** [comment]
yea, was fast and lose, we should tighten it up

File: .gitignore (hunk 1 above)
Comment (webknjaz) [reply .gitignore:1]: Not just per python but per each permutation of python+OS+arch. This is what extensive locking looks like: https://github.com/cherrypy/cheroot/tree/main/requirements. However, to implement such an integrated environment, a lot of effort is required, so I'd postpone.

Comment (jakob-keller) [reply .gitignore:1]: I was aware of that and agree that it should be addressed at some point. As it stands, we could throw out pip-tools from CI/CD and just run `pip install .[awscli,boto3] -r requirements-dev.in` or similar. But I suggest to leave it as it is for now and have a separate effort to improve that aspect, maybe even switch dependency management tool again (poetry, hatch, ...).

**Comment by jakob-keller** [comment]
I've opened #1188 for tracking the step-change improvement in dependency management.

Nonetheless, I believe it makes sense to do this PR now to throw out some technical debt.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**